pysmarthashtag/api/authentication.py

```
@@ -190,7 +301,12 @@ async def _login(self):
                 context = r_context.url.params["context"]
                 _LOGGER.debug("Context: %s", context)
             except KeyError:
-                raise SmartAPIError("Could not get context from login page")
+                # PATCH: include status + body so the breaker can classify (rate-limit vs other).
+                raise SmartAPIError(
+                    "Could not get context from login page "
+                    f"(HTTP {r_context.status_code}, "
+                    f"body={r_context.text[:200]!r})"
+                )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)